### ISSUE TITLE
Wnt to use MURIL for semantic similarilty

### ISSUE THREAD

**Issue by ankitkr3** [UKPLab/sentence-transformers#1280]
Hi @nreimers 

i am trying to use muril for semantic similarily between multiple Hindi sentences, i found the following piece of code to work with :

`model_name = './Downloads/MuRIL_1/'

# Use BERT for mapping tokens to embeddings
word_embedding_model = models.BERT(model_name)

# Apply mean pooling to get one fixed sized sentence vector
pooling_model = models.Pooling(word_embedding_model.get_word_embedding_dimension(),
                               pooling_mode_mean_tokens=True,
                               pooling_mode_cls_token=False,
                               pooling_mode_max_tokens=False)

model = SentenceTransformer(modules=[word_embedding_model, pooling_model])`

But i am getting this error **module 'sentence_transformers.models' has no attribute 'BERT'**

**Comment by nreimers** [comment]
You need to use models.Transformer instead of models.BERT

**Comment by ankitkr3** [comment]
> You need to use models.Transformer instead of models.BERT

Hi @nreimers 

i am using the following code, still getting error to load it finally in the sentenceTransformer


`from transformers import AutoTokenizer, AutoModelForMaskedLM
  
tokenizer = AutoTokenizer.from_pretrained("google/muril-base-cased")

word_embedding_model = models.Transformer(tokenizer)

--Apply mean pooling to get one fixed sized sentence vector
pooling_model = models.Pooling(word_embedding_model.get_word_embedding_dimension(),
                               pooling_mode_mean_tokens=True,
                               pooling_mode_cls_token=False,
                               pooling_mode_max_tokens=False)

model = SentenceTransformer(modules=[word_embedding_model, pooling_model])`

can you please help me with the correct way to load the Muril model?

**Comment by nreimers** [comment]
You must pass the name to models.Transformer, not the tokenizer object

**Comment by ankitkr3** [comment]
@nreimers  it's working, but the cosine similarity is either 0.99 or 1.00 for almost all the sentences whether they are similar or not.

**Comment by ankitkr3** [comment]
@nreimers also i am using **distiluse-base-multilingual-cased-v2** parallelly which is working so far good for Hindi and other languages..

**Comment by nreimers** [comment]
I'm not familiar with MURIL, but training with only MLM does not work. See the SBERT paper.

You need a specific fine-tuning step so that the vector space can be used for embedding comparison. For this, you need text pairs (like Q&A pairs)

**Comment by ankitkr3** [comment]
@nreimers okay thanks, but can i use **distiluse-base-multilingual-cased-v2** ? i believe it should work better.

**Comment by nreimers** [comment]
Yes. Have a look at: https://www.sbert.net/docs/pretrained_models.html#multi-lingual-models
